operation/fhir-operation-member-match/src/main/java/com/ibm/fhir/operation/davinci/hrex/provider/strategy/DefaultMemberMatchStrategy.java

```diff
@@ -250,13 +250,25 @@ public boolean visit(String elementName, int elementIndex, Identifier identifier
          * @param identifier
          */
         private void addIdValue(Identifier identifier) {
-            Uri sys = identifier.getSystem();
-            com.ibm.fhir.model.type.String val = identifier.getValue();
-            if (sys != null && sys.getValue() != null) {
-                this.system = sys.getValue();
-            }
-            if (val != null && val.getValue() != null) {
-                this.value = val.getValue();
+            if (identifier.getType() != null
+                    && identifier.getType().getCoding() != null) {
+                // If there is  more than one Coding it has to be MB.
+                for (Coding coding : identifier.getType().getCoding()) {
+                    if (coding.getSystem() != null
+                            &&  "http://terminology.hl7.org/CodeSystem/v2-0203"
+                                    .equals(coding.getSystem().getValue())
+                            && "MB".equals(coding.getCode().getValue())) {
```

Comment:
technically i think coding.getCode() could be null, but I think its probably rare we'd hit that

Reply:
Yeah - in negative testing